src/BinomialQueue/Internals.hs

```diff
@@ -219,11 +227,17 @@ unions = foldl' union empty
 
 -- | \(O(n)\). Map elements and collect the 'Just' results.
 mapMaybe :: Ord b => (a -> Maybe b) -> MinQueue a -> MinQueue b
-mapMaybe f (MinQueue ts) = mapMaybeQueue f (const empty) empty ts
+mapMaybe f = flip foldlU' empty $ \q a ->
+  case f a of
+    Nothing -> q
+    Just b -> insertEager b q
 
 -- | \(O(n)\). Map elements and separate the 'Left' and 'Right' results.
 mapEither :: (Ord b, Ord c) => (a -> Either b c) -> MinQueue a -> (MinQueue b, MinQueue c)
-mapEither f (MinQueue ts) = mapEitherQueue f (const (empty, empty)) (empty, empty) ts
+mapEither f = (fromPartition .) . flip foldlU' (Partition empty empty) $ \(Partition ls rs) a ->
```

Comment:
Can we please avoid `(f .) . g`? I find that hard to understand.

Reply:
Yeah, I suppose I can expand the whole thing out.

Reply:
Or would you be happy with `(fromPartition .) $ flip ...`?

Reply:
I think it would be better to just expand it to a lambda, that would be easier to understand.